app/controllers/charityInformation/CharityInformationSummaryController.scala

```diff
@@ -52,9 +54,31 @@ class CharityInformationSummaryController @Inject()(
   def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
 
     for {
-      updatedAnswers <- Future.fromTry(result = request.userAnswers.set(Section1Page, true))
+      updatedAnswers <- Future.fromTry(result = checkComplete(request.userAnswers) match {
+        case None => Success(request.userAnswers)
+        case inProgressOrComplete => request.userAnswers.set(Section1Page, inProgressOrComplete.get)
+      })
       _              <- sessionRepository.set(updatedAnswers)
     } yield Redirect(navigator.nextPage(CharityInformationSummaryPage, NormalMode, updatedAnswers))
 
   }
+
+  def checkComplete(userAnswers: UserAnswers): Option[Boolean] = {
+    val pagesAlwaysRequired = Seq(CharityNamePage, CharityContactDetailsPage, CharityOfficialAddressLookupPage, CanWeSendToThisAddressPage)
+    val charityPostalAddressIsDefined = userAnswers.arePagesDefined(Seq(CharityPostalAddressLookupPage))
+
+    Seq(
```

Comment:
could you please do the refactoring as discussed to make it more simpler and readable